File: tests/test-minecraft-osp/src/main/java/pize/tests/voxelgame/server/ServerConfiguration.java
package pize.tests.voxelgame.server;

import pize.tests.voxelgame.main.Version;

public class ServerConfiguration{
    
    private String address;
    private int port;
    private Version version;
    private String defaultLevelName;
    private int maxRenderDistance;
    
    public void loadDefaults(){
        port = 22854;
        address = "0.0.0.0";
        version = new Version();
        defaultLevelName = "world1";
        maxRenderDistance = 12;
    }
    
    public String getAddress(){
        return address;
    }
    
    public int getPort(){
        return port;
    }
    
    public Version getVersion(){
        return version;
    }
    
    public String getDefaultLevelName(){
        return defaultLevelName;
    }
    
    public int getMaxRenderDistance(){
        return maxRenderDistance;
    }
    
}

File: tests/test-minecraft-osp/src/main/kotlin/pize/tests/voxelgame/server/ServerConfiguration.kt
package pize.tests.voxelgame.server

import pize.tests.voxelgame.main.Version

class ServerConfiguration {
    var address: String? = null
        private set
    var port = 0
        private set
    var version: Version? = null
        private set
    var defaultLevelName: String? = null
        private set
    var maxRenderDistance = 0
        private set

    fun loadDefaults() {
        port = 22854
        address = "0.0.0.0"
        version = Version()
        defaultLevelName = "world1"
        maxRenderDistance = 12
    }
}
